src/main/java/org/jfrog/hudson/pipeline/common/types/buildInfo/BuildInfo.java

```diff
@@ -127,7 +127,10 @@ public void setStartDate(Date date) {
 
     @Whitelisted
     public void append(BuildInfo other) {
-        this.deployableArtifacts.addAll(other.deployableArtifacts);
+        // Preserve existing modules if there are duplicates
+        other.deployableArtifactsByModule.putAll(this.deployableArtifactsByModule);
+        this.deployableArtifactsByModule = other.deployableArtifactsByModule;
```

Comment:
I think that if we're appending two build-info's, which both include modules with the same name, these modules should merged. It looks like this code overrides, instead of merges, these modules.
Also, it loos this code is identical to the code in the **appendDeployableArtifactsByModule** method.